### ISSUE TITLE
[V2] Echoserver sample not working with Ingress-Controller V2

### ISSUE THREAD

**Issue by stuzfuz** [kubernetes-sigs/aws-load-balancer-controller#1480]
Hi,

I've created a new EKS-Cluster with Fargate support.
Enabled the OIDC-Provider and deployed the new AWS-ALB-Ingress-Controller Version 2.
Then tried to deploy the 'Echoserver' sample.

The Deployment itself is working, but the ingress is not working and the AWS-ALB-Ingress-Controller is logging the following:
`
{"level":"error","ts":1602237189.6228282,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"ingress","name":"echoserver","namespace":"echoserver","error":"MissingEndpoint: 'Endpoint' configuration is required for this service","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90"}
`

I'm using the following args for the AWS-ALB-Ingress-Controller, as the automatic setup is not working because of an timeout issue:
`
args:
            - --cluster-name=EKS-CLUSTER
            - --aws-vpc-id=vpc-XXXXXX
            - --aws-region=us-east-1
`

Ingress:
`
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: echoserver
  namespace: echoserver
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/subnets: subnet-XXXXX,subnet-XXXXX
    alb.ingress.kubernetes.io/tags: Environment=dev,Team=test
spec:
  rules:
    - host: example.domain.com
      http:
        paths:
          - path: /
            backend:
              serviceName: echoserver
              servicePort: 80
`

Describe and Get Output:

**HOST:/$ kubectl describe ingress -n echoserver**
Name:             echoserver
Namespace:        echoserver
Address:
Default backend:  default-http-backend:80 (<none>)
Rules:
  Host                                   Path  Backends
  ----                                   ----  --------
  example.domain.com
                                         /   echoserver:80 (XX.XX.XX.XX:8080)
Annotations:
  alb.ingress.kubernetes.io/scheme:                  internet-facing
  alb.ingress.kubernetes.io/subnets:                 subnet-XXXX,subnet-XXXXX
  alb.ingress.kubernetes.io/tags:                    Environment=dev,Team=test
  kubectl.kubernetes.io/last-applied-configuration:  {"apiVersion":"extensions/v1beta1","kind":"Ingress","metadata":{"annotations":{"alb.ingress.kubernetes.io/scheme":"internet-facing","alb.ingress.kubernetes.io/subnets":"subnet-XXXXX,subnet-XXXXX","alb.ingress.kubernetes.io/tags":"Environment=dev,Team=test","kubernetes.io/ingress.class":"alb"},"name":"echoserver","namespace":"echoserver"},"spec":{"rules":[{"host":"example.domain.com","http":{"paths":[{"backend":{"serviceName":"echoserver","servicePort":80},"path":"/"}]}}]}}

 kubernetes.io/ingress.class:  alb
Events:                         <none>

**HOST:/$ kubectl get ingress -n echoserver**
NAME         HOSTS                                   ADDRESS   PORTS   AGE
echoserver   example.domain.com             80      168m

**Comment by kishorj** [comment]
need to reproduce the issue

**Comment by stuzfuz** [comment]
Steps I have done:
- eksctl create cluster -n $CLUSTER_NAME --alb-ingress-access --fargate --full-ecr-access  --vpc-private-subnets=subnet-XXX,subnet-XXXX,subnet-XXX --vpc-public-subnets=subnet-XXXX,subnet-XXXX,subnet-XXX
- eksctl utils associate-iam-oidc-provider --cluster $CLUSTER_NAME --approve
- https://github.com/kubernetes-sigs/aws-alb-ingress-controller/blob/v2_ga/config/samples/install_v2_0_0_rc2.yaml <- downloaded and changed the "cluster-name", added "--aws-region=us-east-1" and "--aws-vpc-id=vpc-XXXX"
- Deployed the Echoserver sample (1:1 like in the docs): https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/walkthrough/echoserver/
- Fetched the logs which are attached on my initial post

**Comment by kishorj** [comment]
@stuzfuz Thanks for the steps, I will try it in us-east-1 region.

**Comment by kishorj** [comment]
I tested on us-east-1 region with eksctl auto-generated subnets. The echoserver example worked both with or without the `alb.ingress.kubernetes.io/subnets` annotation. I was able to curl the ALB as follows
```
$ curl -H "Host: echoserver.example.com" k8s-echoserv-echoserv-xxxx-yyyy.us-east-1.elb.amazonaws.com
CLIENT VALUES:
client_address=192.168.xx.xx
command=GET
real path=/
query=nil
request_version=1.1
request_uri=http://echoserver.example.com:8080/
```

Questions
- Are all the subnets tagged appropriately?
```
   kubernetes.io/cluster/$CLUSTER_NAME
   kubernetes.io/role/elb or kubernetes.io/role/internal-elb
```
- Are the subnets across different AZs?
- What is the timeout issue with auto discovery? Did  eksctl fail to create the cluster?

Next steps
Retry the tests on the 2.0.0 release. If you encounter any issues, please share it with us.

**Comment by M00nF1sh** [comment]
Hi @stuzfuz 
Does this issue still exist with our v2.0.0 release? we have tested it works with Fargate.

Feel free to reopen if this issue still exists.

**Comment by stuzfuz** [comment]
Hi @M00nF1sh & @kishorj 

thanks for your replies.
The issue is still there with version 2.0.0 GA.

History of my commands to reproduce it:
`
 1996  eksctl create cluster -n $CLUSTER_NAME --alb-ingress-access --fargate --full-ecr-access  --vpc-private-subnets=subnet-XXXX,subnet-XXXX,subnet-XXXX --vpc-public-subnets=subnet-XXXX,subnet-XXXX,subnet-XXXX
 2005  eksctl utils associate-iam-oidc-provider --region us-east-1 --cluster $CLUSTER_NAME --approve
 2007  curl -o iam-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/install/iam_policy.json
 2008  aws iam create-policy     --policy-name AWSLoadBalancerControllerIAMPolicy     --policy-document file://iam-policy.json
 2009  eksctl create iamserviceaccount --cluster=$CLUSTER_NAME --namespace=kube-system --name=aws-load-balancer-controller --attach-policy-arn=arn:aws:iam::XXXXXXX:policy/AWSLoadBalancerControllerIAMPolicy --approve
 2010  helm repo add eks https://aws.github.io/eks-charts
 2011  kubectl apply -k "github.com/aws/eks-charts/stable/aws-load-balancer-controller//crds?ref=master"
 2012  helm upgrade -i aws-load-balancer-controller eks/aws-load-balancer-controller -n kube-system --set clusterName=$CLUSTER_NAME --set serviceAccount.create=false --set serviceAccount.name=aws-load-balancer-controller
`

Log output:
`
{"level":"info","ts":1603969255.4013326,"msg":"version","GitVersion":"v2.0.0","GitCommit":"1028fa4f363a9a8e37b07ff6a093b7b422923512","BuildDate":"2020-10-21T22:17:18+0000"}
{"level":"error","ts":1603969261.6114073,"logger":"setup","msg":"unable to initialize AWS cloud","error":"failed to introspect region from EC2Metadata, specify --aws-region instead if EC2Metadata is unavailable: EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get \"http://169.254.169.254/latest/dynamic/instance-identity/document\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}
`

**Comment by stuzfuz** [comment]
Please reopen the issue as I am not allowed to.

**Comment by kishorj** [comment]
@stuzfuz Looks like EC2Metadata is either not available or disabled on your environment. In this case, could you specify the --aws-region, and --aws-vpc-id controller flag.
For helm you could specify the following additional parameters, available in helm chart version 1.0.4
--set region=<AWS_REGION> --set vpcId=<AWS_VPC_ID>

We will be able to debug further once we are past this error.

**Comment by M00nF1sh** [comment]
@stuzfuz please create a new issue if the problem still exists.
Also, if EC2Metadata is not accessible from the controller, please check my comment at this issue: https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/1747#issuecomment-752215264
